includes/class-wc-payments-order-success-page.php

```diff
@@ -73,6 +75,58 @@ public function show_woopay_payment_method_name( $payment_method_title, $abstrac
 		return ob_get_clean();
 	}
 
+	/**
+	 * Add the BNPL logo to the payment method name on the order received page.
+	 *
+	 * @param string            $payment_method_title the default payment method title.
+	 * @param WC_Abstract_Order $abstract_order the order being shown.
+	 */
+	public function show_bnpl_payment_method_name( $payment_method_title, $abstract_order ) {
+
+		// Only change the payment method title on the order received page.
+		if ( ! is_order_received_page() ) {
+			return $payment_method_title;
+		}
+
+		$order_id = $abstract_order->get_id();
+		$order    = wc_get_order( $order_id );
+		$bnpl_methods = [
+			'Affirm',
+			'Afterpay',
+			'Clearpay',
+			'Klarna',
+		];
+
+		if ( ! $order || ! in_array( $payment_method_title, $bnpl_methods, true ) ) {
+			return $payment_method_title;
+		}
```

Comment:
```suggestion
		$bnpl_methods = [
			new Affirm_Payment_Method()->get_title(),
			new Afterpay_Payment_Method()->get_title(),
			new Clearpay_Payment_Method()->get_title(),
			new Klarna_Payment_Method()->get_title(),
		];

		if ( ! $order || ! in_array( $payment_method_title, $bnpl_methods, true ) ) {
			return $payment_method_title;
		}
```
This comparison here is a little rigid and has potential to crack. Can we use the title from the `UPE_Payment_Method` class instead? Note this example is just one example of how to gather these titles and might not even be the best one--you can probably get these instantiated classes from one of the static WCPay map utility functions too.